Makefile

```diff
@@ -107,15 +116,38 @@ clientset: ## Generate a typed clientset
 clean: ## Remove all generated files
 	rm -f bazel-*
 
+.PHONY: all-docker-build
+all-docker-build: $(addprefix sub-docker-build-,$(ALL_ARCH))
+	@echo "updating kustomize image patch file for manager resource"
+	hack/sed.sh -i.tmp -e 's@image: .*@image: '"$(CONTROLLER_IMG):$(TAG)"'@' ./config/default/manager_image_patch.yaml
+
+sub-docker-build-%:
+	$(MAKE) ARCH=$* docker-build
+
 .PHONY: docker-build
 docker-build: generate fmt vet manifests ## Build the docker image for controller-manager
-	docker build . -t ${CONTROLLER_IMG}
+	docker build --build-arg ARCH=$(ARCH) . -t $(CONTROLLER_IMG)-$(ARCH):$(TAG)
 	@echo "updating kustomize image patch file for manager resource"
-	hack/sed.sh -i.tmp -e 's@image: .*@image: '"${CONTROLLER_IMG}"'@' ./config/default/manager_image_patch.yaml
+	hack/sed.sh -i.tmp -e 's@image: .*@image: '"${CONTROLLER_IMG}-$(ARCH):$(TAG)"'@' ./config/default/manager_image_patch.yaml
+
+.PHONY:all-push ## Push all the architecture docker images and fat manifest docker image
+all-push: all-docker-push push-manifest
+
+.PHONY:all-docker-push ## Push all the architecture docker images
+all-docker-push: $(addprefix sub-docker-push-,$(ALL_ARCH))
+
+sub-docker-push-%:
+	$(MAKE) ARCH=$* docker-push
+
+.PHONY: push-manifest
```

Comment:
I think adding a comment linking to the bazel issue for adding manifest list support would be good, that way we can track when that is merged and update the build tooling to build the manifest list using bazel when it has: https://github.com/bazelbuild/rules_docker/issues/300

Reply:
👍